Windows/CommonDialog/MainWindow.xaml.cs

```diff
@@ -0,0 +1,166 @@
+            InitializeComponent();
+            DataContext = new DataModel();
+        }
+
+        private void Button_OpenFileDialog(object sender, RoutedEventArgs e)
+        {
+            // Instantiate the open file dialog box.
+            // Setting the following properties are optional
+            OpenFileDialog openFileDialog = new OpenFileDialog()
+            {
+                Title = "Select any text file...",
+                InitialDirectory = @"C:\Windows",
+                Filter = "Text files (*.txt)|*.txt"
+            };
+
+            // Forces the preview pane to be shown irrespective of the system settings
+            //openFileDialog.ForcePreviewPane = true;
+
+            // Allows to select multiple files in the dialog
+            openFileDialog.Multiselect = true;
+
+
+            string text = "";
+
+            // If the user clicks OK button on the dialog, the ShowDialog() method returns true
+            // otherwise it returns false.
+            if (openFileDialog.ShowDialog() == true)
+            {
+                var streams = openFileDialog.OpenFiles();
+                foreach (var stream in streams)
+                {
+                    using (stream)
+                    {
+                        using (var reader = new System.IO.StreamReader(stream))
+                        {
+                            text += reader.ReadToEnd();
+                        }
+                        text += "\n\n--- END OF FILE ---\n";
+                    }
+                }
+            }
+            else
+            {
+                text = "No file selected. \nEither cancel button was clicked or dialog was closed";
+            }
+
+            // Setting the data context of this application. Not related to dialog functionality.
+            DataModel data = DataModel.GetDialogRelatedInfo(openFileDialog);
+            data.ResultBody = text;
+            DataContext = data;
+        }
+
+        private void Button_SaveFileDialog(object sender, RoutedEventArgs e)
+        {
+            SaveFileDialog saveFileDialog = new SaveFileDialog()
+            {
+                Title = "Select file to save ...",
+                InitialDirectory = Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments),
+                Filter = "Text files (*.txt)|*.txt"
+            };
+
+            // Setting this flag to true will prompt the user if the file is going to be created.
+            saveFileDialog.CreatePrompt = true;
+
+            // Setting this flag to true will prompt the user if the file is going to be overwritten.
+            saveFileDialog.OverwritePrompt = true;
+
+            // Setting this flag to true will restore the Application.CurrentDirectory.
+            //saveFileDialog.RestoreDirectory = true;
+
+            // We can attach an event handler to the FileOk event of the dialog box.
+            saveFileDialog.FileOk += SaveFileDialog_FileOk;
+
+            string text = "";
```

Comment:
This does not need to be initialized to `""` I believe. You could do `StringBuilder` again, since you add couple of lines.